app/code/Magento/SendFriend/Block/Send.php

```diff
@@ -15,69 +24,57 @@
  * @author      Magento Core Team <core@magentocommerce.com>
  * @since 100.0.2
  */
-class Send extends \Magento\Framework\View\Element\Template
+class Send extends Template
 {
     /**
      * SendFriend data
      *
-     * @var \Magento\SendFriend\Helper\Data
+     * @var SendFriendHelper
      */
     protected $_sendfriendData = null;
 
     /**
      * Core registry
      *
-     * @var \Magento\Framework\Registry
+     * @var Registry
      */
     protected $_coreRegistry = null;
 
     /**
-     * @var \Magento\Customer\Model\Session
+     * @var CustomerSession
      */
     protected $_customerSession;
 
     /**
-     * @var \Magento\Framework\App\Http\Context
-     */
-    protected $httpContext;
-
-    /**
-     * @var \Magento\Customer\Helper\View
+     * @var CustomerViewHelper
      */
     protected $_customerViewHelper;
 
     /**
-     * @var \Magento\SendFriend\Model\SendFriend
-     */
-    protected $sendfriend;
-
-    /**
-     * @param \Magento\Framework\View\Element\Template\Context $context
-     * @param \Magento\Customer\Model\Session $customerSession
-     * @param \Magento\SendFriend\Helper\Data $sendfriendData
-     * @param \Magento\Framework\Registry $registry
-     * @param \Magento\Customer\Helper\View $customerViewHelper
-     * @param \Magento\Framework\App\Http\Context $httpContext
-     * @param \Magento\SendFriend\Model\SendFriend $sendfriend
+     * @param TemplateContext $context
+     * @param CustomerSession $customerSession
+     * @param SendFriendHelper $sendfriendData
+     * @param Registry $registry
+     * @param CustomerViewHelper $customerViewHelper
+     * @param HttpContext $httpContext
+     * @param SendFriend $sendfriend
      * @param array $data
      */
     public function __construct(
-        \Magento\Framework\View\Element\Template\Context $context,
-        \Magento\Customer\Model\Session $customerSession,
-        \Magento\SendFriend\Helper\Data $sendfriendData,
-        \Magento\Framework\Registry $registry,
-        \Magento\Customer\Helper\View $customerViewHelper,
-        \Magento\Framework\App\Http\Context $httpContext,
-        \Magento\SendFriend\Model\SendFriend $sendfriend,
+        TemplateContext $context,
+        CustomerSession $customerSession,
+        SendFriendHelper $sendfriendData,
+        Registry $registry,
+        CustomerViewHelper $customerViewHelper,
```

Comment:
Why not promote these properties?
```suggestion
        protected readonly CustomerSession $_customerSession,
        protected readonly SendFriendHelper $_sendfriendData,
        protected readonly Registry $_coreRegistry,
        protected readonly CustomerViewHelper $_customerViewHelper,
```